src/provider.tsx

```diff
@@ -0,0 +1,268 @@
+  const {
+    item: stacGeoparquetItem,
+    loading: stacGeoparquetItemLoading,
+    error: stacGeoparquetItemError,
+  } = useStacGeoparquetItem(stacGeoparquetItemId, parquetPath, connection);
+
+  useEffect(() => {
+    function handlePopState() {
+      setHref(new URLSearchParams(location.search).get("href") ?? "");
+    }
+
+    window.addEventListener("popstate", handlePopState);
+    return () => {
+      window.removeEventListener("popstate", handlePopState);
+    };
+  }, []);
+
+  useEffect(() => {
+    selectedCollectionsDispatch({ type: "deselect-all-collections" });
+    if (href) {
+      if (new URLSearchParams(location.search).get("href") != href) {
+        history.pushState(null, "", "?href=" + href);
+      }
+    }
+  }, [href]);
+
+  useEffect(() => {
+    // It should never be more than 1.
+    if (fileUpload.acceptedFiles.length == 1) {
+      setHref(fileUpload.acceptedFiles[0].name);
+    }
+  }, [fileUpload.acceptedFiles]);
+
+  useEffect(() => {
+    if (valueError) {
+      createToastError("Error when getting a STAC value", valueError);
+    }
+  }, [valueError]);
+
+  useEffect(() => {
+    if (collectionsError) {
+      createToastError("Error when getting STAC collections", collectionsError);
+    }
+  }, [collectionsError]);
+
+  useEffect(() => {
+    if (duckDbConnectionError) {
+      createToastError(
+        "Error loading DuckDB",
+        duckDbConnectionError.toString(),
+      );
+    }
+  }, [duckDbConnectionError]);
+
+  useEffect(() => {
+    if (stacGeoparquetTableError) {
+      createToastError(
+        "Error reading stac-geoparquet",
+        stacGeoparquetTableError,
+      );
+    }
+  }, [stacGeoparquetTableError]);
+
+  useEffect(() => {
+    if (stacGeoparquetMetadataError) {
+      createToastError(
+        "Error reading stac-geoparquet metadata",
+        stacGeoparquetMetadataError,
+      );
+    }
+  }, [stacGeoparquetMetadataError]);
+
+  useEffect(() => {
+    if (stacGeoparquetItemError) {
+      createToastError(
+        "Error getting stac-geoparquet item",
+        stacGeoparquetItemError,
+      );
+    }
+  }, [stacGeoparquetItemError]);
+
+  const contextValue = {
+    href,
+    setHref,
+    fileUpload,
+
+    value,
+    valueLoading,
+
+    collections,
+    collectionsLoading,
+    selectedCollections,
+    selectedCollectionsDispatch,
+
+    stacGeoparquetTable,
+    stacGeoparquetMetadata,
+    stacGeoparquetTableLoading,
+    stacGeoparquetMetadataLoading,
+    setStacGeoparquetItemId,
+    stacGeoparquetItem,
+    stacGeoparquetItemLoading,
+  };
+
+  return (
+    <StacMapContext.Provider value={contextValue}>
+      {children}
+      {collections && (
+        <SelectedCollectionsActionBar
+          collections={collections.filter((collection) =>
+            selectedCollections.has(collection.id),
+          )}
+          dispatch={selectedCollectionsDispatch}
+        ></SelectedCollectionsActionBar>
+      )}
+    </StacMapContext.Provider>
+  );
+}
```

Comment:
This Provider is kind of a monster, but I think that's just how frontend apps work? 🤷🏼